### ISSUE TITLE
velero is unable to prune PartiallyFailed backup from object storage

### ISSUE THREAD

**Issue by jhuisss** [vmware-tanzu/velero#5940]
**What steps did you take and what happened:**
I'm using velero 1.9.5, there is a PartiallyFailed backup (restic backup with nfs pv), which results in 4.1G disk usage on the minio server. When I use `velero backup delete <partiallyfailed backup>` cmd to delete the backup, it did delete the backup CR, but it won't release disk space in bucket in the object storage, even though I have set`default-restic-prune-frequency=0h30m0s` and make sure in the resticrepository, the `maintenanceFrequency` equals to 30m0s.
```
# kubectl -n velero describe resticrepository testmypv-large-ns-default-pb6mq
Name:         testmypv-large-ns-default-pb6mq
Namespace:    velero
Labels:       velero.io/storage-location=default
              velero.io/volume-namespace=testmypv-large-ns
Annotations:  <none>
API Version:  velero.io/v1
Kind:         ResticRepository
Metadata:
  Creation Timestamp:  2023-03-02T07:50:13Z
  Generate Name:       testmypv-large-ns-default-
  Generation:          4
  Managed Fields:
    API Version:  velero.io/v1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:generateName:
        f:labels:
          .:
          f:velero.io/storage-location:
          f:velero.io/volume-namespace:
      f:spec:
        .:
        f:backupStorageLocation:
        f:maintenanceFrequency:
        f:resticIdentifier:
        f:volumeNamespace:
      f:status:
        .:
        f:lastMaintenanceTime:
        f:phase:
    Manager:         velero-server
    Operation:       Update
    Time:            2023-03-02T07:50:15Z
  Resource Version:  5610009
  UID:               db87d044-2d57-43e6-9763-376abb727c4d
Spec:
  Backup Storage Location:  default
  Maintenance Frequency:    30m0s
  Restic Identifier:        s3:https://***:9000/restic7.xkou/restic/testmypv-large-ns
  Volume Namespace:         testmypv-large-ns
Status:
  Last Maintenance Time:  2023-03-02T08:23:39Z
  Phase:                  Ready
Events:                   <none>
```
```
root@minio:/minio/data# du -h -d 1
16K	./lost+found
2.9M	./.minio.sys
4.0K	./second-bucket.xkou
4.0K	./first-bucket.xkou
28K	./restic2.xkou
4.1G	./restic7.xkou.          <----- after 1 hour, it is still 4.1G, expect to be 4.0K as empty dir.
4.1G	.
```

**What did you expect to happen:**
When running cmd `velero backup delete <partiallyfailed backup>`, after half an hour, the disk space used by this backup should be released.

**The following information will help us better understand what's going on**:
One wired thing is that deleting `Completed` backup will release the disk used in objectstorage, but to `PartiallyFailed` backup, it won’t release disk space.

BTW, Only 1 backup in this namespace and this object storage bucket, no incremental  backup, so delete the backup should release all disk usage in this bucket.

**Anything else you would like to add:**
velero logs during running `velero backup delete <partiallyfailed backup>`:
```
time="2023-03-02T01:50:58Z" level=info msg="Starting to check for items in namespace" logSource="internal/delete/delete_item_action_handler.go:95" namespace=testmypv-large-ns
time="2023-03-02T01:50:58Z" level=info msg="invoking DeleteItemAction plugins" item=pvc-test1 logSource="internal/delete/delete_item_action_handler.go:111" namespace=testmypv-large-ns
time="2023-03-02T01:50:58Z" level=info msg="Skipping PVCDeleteItemAction for PVC testmypv-large-ns/pvc-test1, PVC does not have a vSphere BackupItemAction snapshot." backup=br-e2e-wc1-backup cmd=/plugins/velero-plugin-for-vsphere controller=backup-deletion deletebackuprequest=velero/br-e2e-wc1-backup-lzpck logSource="/go/src/github.com/vmware-tanzu/velero-plugin-for-vsphere/pkg/plugin/delete_pvc_action_plugin.go:57" pluginName=velero-plugin-for-vsphere
time="2023-03-02T01:50:58Z" level=info msg="Removing PV snapshots" backup=br-e2e-wc1-backup controller=backup-deletion deletebackuprequest=velero/br-e2e-wc1-backup-lzpck logSource="pkg/controller/backup_deletion_controller.go:278"
time="2023-03-02T01:50:58Z" level=info msg="Removing restic snapshots" backup=br-e2e-wc1-backup controller=backup-deletion deletebackuprequest=velero/br-e2e-wc1-backup-lzpck logSource="pkg/controller/backup_deletion_controller.go:303"
time="2023-03-02T01:50:58Z" level=info msg="Removing backup from backup storage" backup=br-e2e-wc1-backup controller=backup-deletion deletebackuprequest=velero/br-e2e-wc1-backup-lzpck logSource="pkg/controller/backup_deletion_controller.go:311"
time="2023-03-02T01:50:58Z" level=info msg="Removing restores" backup=br-e2e-wc1-backup controller=backup-deletion deletebackuprequest=velero/br-e2e-wc1-backup-lzpck logSource="pkg/controller/backup_deletion_controller.go:317"
```

**Environment:**

- Velero version (use `velero version`):  1.9.5
- Velero features (use `velero client config get features`): 
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by Lyndon-Li** [comment]
@jhuisss Please also share the podVolumeBackup CRs included in the partially failed backup

**Comment by Lyndon-Li** [comment]
This looks like a bug of handling Restic snapshot erroneously:

- When restic backup encounter this kind of error, it just prints the error to terminal
- However, restic doesn't stop half-way, instead it continues the backup
- Finally, restic saves the snapshot into backup store

However, on Velero side:

- When restic backup command returns, Velero checks the output from restic
- When Velero sees the error, it treats it as a fatal error and abort
- Therefore the restic snapshot is not retrieved and recorded by PVB
- As a result, the snapshot and its data is lost forever

**Comment by Lyndon-Li** [comment]
Reproduce steps:

- Deploy a pod which has a PV mounted
- Login the pod and cd to mounted volume
- Generate 5 files with each file has 10G data: dd if=/dev/urandom bs=1024M count=10 iflag=fullblock of=tmpfilea
- Create velcro backup with parameters to include the namespace of above pod
- During the backup process, e.g. after 20s of the creation beginning, login the pod again and cd to mounted volume.
- Delete 3 files.
- Wait until velcro backup to finish which should be end with partial failure.

**Comment by Lyndon-Li** [comment]
This problem also happens with Kopia uploader

**Comment by Lyndon-Li** [comment]
After double confirm from the code and testing, Kopia path doesn't have this problem because the snapshot for Kopia path is saved by Velero, if Kopia uploader reports any error, Velero will abort saving the snapshot, so the backed up data will be removed by Kopia GC some time later.

Since in v1.12 and later, we will suppress Restic path, this issue is not with high priority.

**Comment by Lyndon-Li** [comment]
@sseago 
Yes, you are right. Here by `suppress`, I meant we will set Restic path non-default in v1.12.

**Comment by ctrought** [comment]
also experience this volume backups using CSI snapshots with datamover, after the BackupDeleteRequest is completed the objects for the volume backup remain in the S3 bucket.

**Comment by tomerschwartz24** [comment]
happens with kopia too, i deleted the backup resource in velero but data remain in s3 bucket, how do i overcome this?
